frontstage_api/controllers/case_controller.py

```diff
@@ -74,3 +91,57 @@ def post_case_event(case_id, party_id, category, description):
         logger.error('Failed to post to case service', case_id=case_id)
         raise ApiError(url, response.status_code)
     logger.debug('Successfully posted case event', case_id=case_id)
+
+
+def check_case_permissions(party_id, case_party_id, case_id=None):
+    logger.debug('Party requesting access to case', party_id=party_id, case_id=case_id, case_party_id=case_party_id)
+    if party_id != case_party_id:
+        logger.warning('Party does not have permission to access case', party_id=party_id, case_id=case_id, case_party_id=case_party_id)
+        raise NoSurveyPermission(party_id, case_id, case_party_id)
+
+    logger.debug('Party has permission to access case', party_id=party_id, case_id=case_id, case_party_id=case_party_id)
+
+
+def build_full_case_data(case, status_filter=None):
+    logger.debug('Attempting to build survey data', case_id=case['id'])
+    collection_exercise_id = case["caseGroup"]["collectionExerciseId"]
+    collection_exercise = collection_exercise_controller.get_collection_exercise(collection_exercise_id)
+    collection_exercise_formatted = format_collection_exercise_dates(collection_exercise)
+
+    business_party_id = case['caseGroup']['partyId']
+    business_party = party_controller.get_party_by_business_id(business_party_id)
+    survey_id = collection_exercise['surveyId']
+    survey = survey_controller.get_survey(survey_id)
+    collection_instrument_size = collection_instrument_controller.get_collection_instrument_size(case['collectionInstrumentId'])
+    survey_data = {
+        "case": case,
+        "collection_exercise": collection_exercise_formatted,
+        "business_party": business_party,
+        "survey": survey,
+        "collection_instrument_size": collection_instrument_size,
+        'status': status_filter
+    }
+    logger.debug('Successfully built survey data', case_id=case['id'])
+    return survey_data
+
+
+def calculate_case_status(case):
+    case_events = case.get('caseEvents')
```

Comment:
Do this and the function below need logging?

Reply:
Can't hurt to add debug level logging to them anyway